sympy/combinatorics/rewritingsystem.py

```diff
@@ -34,14 +34,16 @@ def __init__(self, group):
 
         # Reduction automaton
         self.reduction_automaton = None
-        self.new_rules = {}
+        self._new_rules = {}
 
         # dictionary of reductions
         self.rules = {}
         self.rules_cache = deque([], 50)
         self._init_rules()
 
-        # Automaton variables
+        # Create a finite state machine as an instance of the StateMachine object
+        self.reduction_automaton = StateMachine('fsm')
+        self.reduction_automaton.add_state('start', state_type='s')
```

Comment:
Did you think about adding the start state in the `StateMachine` class during `__init__`? Any state machine will want to have a start state anyway.

Reply:
This is done inside the `__init__` method itself.

Reply:
Okay 👍